app/controllers/queue_items_controller.rb

```diff
@@ -11,6 +11,18 @@ def create
     redirect_to my_queue_path
   end
 
+  def update_queues
+    update_reviews(params[:queue_ids_with_ratings]) if params[:queue_ids_with_ratings]
+    queue_ids_with_positions = params[:queue_ids_with_positions]
+    if queue_items_have_same_positions?(queue_ids_with_positions)
+      flash[:error] = 'Items cannot have same positions.'
+    else
+      flash[:error] = QueueItem.save_and_update_positions(queue_ids_with_positions)
+    end
```

Comment:
I am not fully understanding this logic -

```ruby
    if queue_items_have_same_positions?(queue_ids_with_positions)
       flash[:error] = 'Items cannot have same positions.'
     else
       flash[:error] = QueueItem.save_and_update_positions(queue_ids_with_positions)
     end
```

If queue items dont have same positions - they will be saved and updated positions but the result will be set to `flash[:error]` ?

```ruby
if some_update
  happy path - i.e. redirect, set flash success/notice
else
  unhappy path - i.e. render template, set flash alert
end
```

You have to learn this pattern - handle happy path first.